app/src/main/java/ch/heigvd/pro/b04/android/Poll/PollActivity.java

```diff
@@ -1,28 +1,38 @@
 package ch.heigvd.pro.b04.android.Poll;
 
+import android.content.Intent;
+import android.os.Bundle;
+
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.lifecycle.ViewModelProvider;
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 
-import android.content.Intent;
-import android.os.Bundle;
-
+import ch.heigvd.pro.b04.android.Home.Home;
 import ch.heigvd.pro.b04.android.Poll.Question.QuestionActivity;
 import ch.heigvd.pro.b04.android.Poll.Question.QuestionAdapter;
 import ch.heigvd.pro.b04.android.R;
+import ch.heigvd.pro.b04.android.Utils.Exceptions.TokenNotSetException;
+import ch.heigvd.pro.b04.android.Utils.Persistent;
 
 public class PollActivity extends AppCompatActivity {
     private PollViewModel state;
 
-        @Override
+    @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_poll);
         Intent intent = getIntent();
         String idPoll = intent.getStringExtra("idPoll");
         String idModerator = intent.getStringExtra("idModerator");
-        String token = intent.getStringExtra("token");
+
+        String token = null;
+        try {
+            token = Persistent.getStoredTokenOrError(getApplicationContext());
+        } catch (TokenNotSetException e) {
+            // No token was found so go back to main activity
+            startActivity(new Intent(this, Home.class));
```

Comment:
Call `finish()` to stop the current activity, instead of adding something on the call stack.